src/tcp_server.rs

```diff
@@ -17,12 +14,38 @@ use crate::ServerMockerError::{
     UnableToWriteTcpStream,
 };
 
+// FIXME: consider consolidating options for both TCP & UDP
+/// Options for the TCP server mocker
+#[derive(Debug, Clone)]
+pub struct TcpMockerOptions {
+    /// Socket address on which the server will listen. Will be set to `127.0.0.1:0` by default.
+    pub socket_addr: SocketAddr,
```

Comment:
Hmmm... TBH I would prefer the socket is only able to listen on loopback. What do you think about it?

Reply:
It seems a bit arbitrary, especially because exposing exposing this allows ip v6. Plus this is similar to the Mockito crate, which also allows arbitrary socket. In short, I think there is not much point to introduce a limit, but it does make sense to have a good default

Reply:
Ok :+1: